src/olympia/addons/models.py

```diff
@@ -1039,16 +1039,19 @@ def attach_listed_authors(addons, addon_dict=None):
     @staticmethod
     def attach_previews(addons, addon_dict=None, no_transforms=False):
         if addon_dict is None:
-            addon_dict = dict((a.id, a) for a in addons)
+            addon_dict = {a.id: a for a in addons}
 
         qs = Preview.objects.filter(addon__in=addons,
                                     position__gte=0).order_by()
         if no_transforms:
             qs = qs.no_transforms()
         qs = sorted(qs, key=lambda x: (x.addon_id, x.position, x.created))
-        for addon, previews in itertools.groupby(qs, lambda x: x.addon_id):
-            addon_dict[addon].all_previews = list(previews)
-        # FIXME: set all_previews to empty list on addons without previews.
+        for addon_id, previews in itertools.groupby(qs, lambda x: x.addon_id):
+            addon_dict[addon_id]._all_previews = list(previews)
+            addon_dict[addon_id] = None
+        # set _all_previews to empty list on addons without previews.
```

Comment:
Can you add a test for this?

Reply:
https://github.com/mozilla/addons-server/pull/8948/commits/5ea6540092e247a3a22cb31764dc3bf956c6771a